cli/lsp/parent_process_checker.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use tokio::time::sleep;
+use tokio::time::Duration;
+
+/// Starts a task that will check for the existence of the
+/// provided process id. Once that process no longer exists
+/// it will terminate the current process.
+pub fn start(parent_process_id: u32) {
+  tokio::task::spawn(async move {
+    loop {
+      sleep(Duration::from_secs(30)).await;
+
+      if !is_process_active(parent_process_id) {
+        eprintln!("Parent process lost. Exiting.");
+        std::process::exit(1);
+      }
+    }
+  });
+}
+
+#[cfg(unix)]
+fn is_process_active(process_id: u32) -> bool {
+  unsafe {
+    // signal of 0 checks for the existence of the process id
+    libc::kill(process_id as i32, 0) == 0
+  }
+}
```

Comment:
I'm not really familiar with libc or winapi, so I'm not sure if what's being done here is correct. Is there a better way of doing this? I did not want to pull in another dependency and didn't see a way to do this with our existing dependencies, but I am likely missing something.

Reply:
This is the only portable way as far as I know.